src/ast/codegen_llvm.cpp

```diff
@@ -758,7 +758,26 @@ void CodegenLLVM::visit(Call &call)
     }
     expr_ = stackid;
   }
-
+  else if (call.func == "strncmp") {
+    uint64_t size = static_cast<Integer *>(call.vargs->at(2))->n;
+    const auto& left_arg = call.vargs->at(0);
+    const auto& right_arg = call.vargs->at(1);
+
+    if (right_arg->is_literal) {
```

Comment:
Is there a reason these literal checks are required for the `==` string check? This version at least seems to work without them. Is it to improve performance on an inline string match?

Reply:
Yeah, this is an optimization so we don't have to save the literal string on memory just to load it later. `==` would work without it as well. I suggest keeping it.